.github/workflows/ci.yml

```diff
@@ -0,0 +1,16 @@
+name: macOS
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: macos-latest
+    steps:
+    # Checks-out the repo. More at: https://github.com/actions/checkout
+    - uses: actions/checkout@v2
+    - name: Select Xcode version
+      run: sudo xcode-select -switch /Applications/Xcode_11.4.app
+    - name: Build project
+      run: swift build
+    - name: Run tests
+      run: swift test
```

Comment:
What is the oldest OS this project supports? We should be running tests against that SDK if possible.

Reply:
@dfed We only get macOS 15 from GitHub actions, so I think running on the latest versions makes sense here https://help.github.com/en/actions/reference/software-installed-on-github-hosted-runners

Reply:
Got it. @fdiaz can we use GitHub actions to build on an iOS simulator? Or do we need a 3rd party integration for that?

Reply:
We have control on what command we run and we have Xcode installed, so we should be able to build using `xcodebuild`